### PR TITLE
Add error boundaries and API error handling

### PR DESCRIPTION
**Pull request by LanesGood**
This PR:
- adds an ErrorBoundary component around the map and summary panel individually so that one component crashing does not crash the whole application
-  Add app/error.tsx for route error boundary for main content area, replacing generic `NextError` page with one wrapped by the app
- Extracts `handleApiError` and registers it as an Axios response interceptor on the shared `api` instance: 401 clears auth tokens and reloads the page; 5xx emits a `console.error` with status and URL.

## Test plan

- [x] Temporarily add `throw new Error("test")` at the top of `MainMap` — confirm the map area shows the boundary fallback while the summary panel and control panel remain functional
- [x] Same for `SummaryPanel` — confirm the map stays visible and the "Reset filters" button clears active filters in the control panel
- [x] Manually remove the `token` key from localStorage while logged in and trigger an API call — confirm remaining tokens are cleared and the header reflects logged-out state
- [ ] Trigger a 5xx from the network tab and confirm `[API] Server error` appears in the console

🤖 Generated with Claude Code

Contributes to #150

### CODE REVIEW

File: src/utils/api.ts [hunk 1]
```diff
@@ -12,3 +12,23 @@ export const api = axios.create({
   baseURL: API_ENDPOINT,
   timeout: 30000,
 });
+
+export function handleApiError(error: {
+  response?: { status: number };
+  config?: { url?: string };
+}): Promise<never> {
+  if (error.response?.status === 401) {
+    localStorage.removeItem('token');
```
Comment (hanbyul-here) [inline src/utils/api.ts]: I don't understand why the tokens need to be cleared when 401 error happens - This will log the user out immediately - is this what we want?

Comment (LanesGood) [reply src/utils/api.ts]: @hanbyul-here I think this is more about not caching auth when the username/password results in an incorrect/non-authenticated user. I think a 401 would only occur if a token expired or if a user's authentication changes on the backend (aka, if an admin removes a user's credentials.) 

Though I just tested this out on the live platform - and it doesn't seem to correspond with the removal a user's credentials. In that case, the platform should look for error status 403, and then log the user out...

**Comment by hanbyul-here** [comment]
I added bunch of small things
 - for auth related error (which I expect the code to be 403? - need to confirm) - Show toaster to remind users to log out and log in again. 
 - I made react query to throw on error - this should trigger error boundary component to be rendered if there is no early catch (such as summary panel rows)
 - I made download page to treat each endpoint independently, render the datasets even when one of the endpoint fails (This probably should have been handled in this PR: https://github.com/developmentseed/moz-proenergia-web/pull/154/changes )
 - I made explorer to function when contextual layer load fails (toaster will display saying that layer load fails)

**Review by hanbyul-here — APPROVED** [review]
@LanesGood please test this change to see if it works as expected before merge

**Comment by LanesGood** [comment]
Thanks for all your work on this @hanbyul-here! I can confirm that the majority of the error handling is now working even better and prevents issues.

There is one issue I see with a layer that may be deleted on the backend; if you try to load
https://deploy-preview-153--proenergina.netlify.app/model/productive-use-economic-development/?layers=10vector-source

You can see that the map crashes and the component text reads `Cannot read properties of null (reading 'replace')`. I'm not seeing any network errors, though I do see that the `raw_file` property for this layer appears to be null. Shouldn't this be caught in the "Make explorer work even when contextual layer fails" catch?

Edit: looks like the layer in question was disabled (set to not approved): 
image
So shouldn't it not show up? Is it perhaps cached as part of the model response?

**Comment by hanbyul-here** [comment]
🤔 the error boundary that was put is at the network request level - this error is from when network requests succeeded, but the data is not as expected. I didn't expect that there would be any case that raw_file is null - (The error is from `deriveSource` level - where the filePath cannot be null). I am checking in with backend to see if this was a hiccup from backend or not. but I will also add the logics to filter out the layers without any raw_file value.

**Comment by LanesGood** [comment]
Thanks @hanbyul-here! I've tested with the tests above, and everything is working as expected. Merging.
